### PR TITLE
Docker fix tesseract module location

### LINKED ISSUE #378: docker: mixing preinstalled and user-downloaded resources
In https://ocr-d.de/en/models we have this paragraph: 

> To download models to `./models` in the host FS and `/usr/local/share/ocrd-resources` in Docker:
> 
> ```sh
> docker run --user $(id -u) \
>   --volume $PWD/models:/usr/local/share/ocrd-resources \
> ocrd/all \
> ocrd resmgr download ocrd-tesserocr-recognize eng.traineddata\; \
> ocrd resmgr download ocrd-calamari-recognize default\; \
> ...
> ```
> 
> To run processors, as usual do:
> 
> ```sh
> docker run --user $(id -u) --workdir /data \
>   --volume $PWD/data:/data \
>   --volume $PWD/models:/usr/local/share/ocrd-resources \
>   ocrd/all ocrd-tesserocr-recognize -I IN -O OUT -P model eng
> ```
> 
> This principle applies to all `ocrd/*` Docker images, e.g. you can replace `ocrd/all` above with `ocrd/tesserocr` as well.

@kba came up with the question if that is still correct for `ocrd-tesserocr-*`: 
https://github.com/OCR-D/ocrd-website/pull/348/files/daf4fce2135f58b4f4ba43ef30e375e161e89651#r1159502451

### PR DESCRIPTION
**Pull request by bertsky**
fixes #378

### CODE REVIEW

**Comment by bertsky** [comment]
CI: see [here](https://github.com/bertsky/ocrd_all/actions/runs/5361185894/jobs/9726892882).

(Still unclear why regular CircleCI does not test PRs. Perhaps we should remove the branch filter?)

Ready for review!

File: Makefile [hunk 1]
```diff
@@ -725,6 +729,8 @@ show:
 	@echo OCRD_MODULES = $(OCRD_MODULES)
 	@echo OCRD_EXECUTABLES = $(OCRD_EXECUTABLES:$(BIN)/%=%)
 
+show-%: ; @echo $($*)
```
Comment (kba) [inline Makefile:732]: Neat trick.

File: Dockerfile [hunk 2]
```diff
@@ -116,8 +119,20 @@ RUN ldconfig
...
+# to mount for model persistence; with named volumes, the preinstalled models
+# will be copied to the host and complemented by downloaded models; tessdata
+# is the only problematic module location
+RUN mkdir -p  $XDG_DATA_HOME/tessdata
+RUN mv $XDG_DATA_HOME/tessdata $XDG_CONFIG_HOME/ocrd-tesserocr-recognize
+RUN ln -s $XDG_CONFIG_HOME/ocrd-tesserocr-recognize $XDG_DATA_HOME/tessdata
+
+# finally, alias/symlink all ocrd-resources to /models for shorter mount commands
+RUN mkdir -p $XDG_CONFIG_HOME
+RUN mv $XDG_CONFIG_HOME /models && ln -s /models $XDG_CONFIG_HOME
```
Comment (kba) [inline Dockerfile:135]: Much cleaner, in addition to solving the `tessdata` issue.

**Review by kba — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -319,9 +319,10 @@ To use the built executables, simply activate the virtual environment:
     ocrd --help
     ocrd-...
 
-For the Docker image, run it with your data path mounted as a user:
+For the Docker image, run it with your data path mounted as a user,
+and the processor resources as named volume (for model persistency):
 
-    docker run -it -u $(id -u):$(id -g) -v $PWD:/data ocrd/all
+    docker run -it -u $(id -u):$(id -g) -v $PWD:/data -v ocrd-models:/models ocrd/all
```
Comment (kba) [inline README.md:325]: @lena-hinrichsen Which means we can and must adopt the documentation again to use that pattern consistently.

Comment (bertsky) [reply README.md:325]: @kba like I [said](https://github.com/OCR-D/ocrd_all/issues/378#issuecomment-1603157185), https://github.com/OCR-D/ocrd-website/pull/357 already covers that (and much more).

**Comment by MehmedGIT** [comment]
Unfortunately, I cannot confirm that this works... 

```shell
SIF_PATH="/scratch1/users/${USER}/ocrd_all_maximum_image.sif"
OCRD_MODELS_DIR="/scratch1/users/${USER}/ocrd_models"
OCRD_MODELS_DIR_IN_DOCKER="/models"

singularity exec --bind "${OCRD_MODELS_DIR}:${OCRD_MODELS_DIR_IN_DOCKER}" "${SIF_PATH}" ocrd resmgr download ocrd-tesserocr-recognize '*'
```

Does not work. However, if I do it the old way I used to with `OCRD_MODELS_DIR_IN_DOCKER="/usr/local/share"` then it works but effectively overwrites that path which leads to other issues such as https://github.com/OCR-D/ocrd_olena/issues/87

**Pull request merged**